relay-metrics/src/aggregation.rs

```diff
@@ -1055,6 +1074,10 @@ impl Aggregator {
         let timestamp = key.timestamp;
         let project_key = key.project_key;
 
+        let key = key
+            .validate_name(&self.config)
+            .map_err(|()| AggregateMetricsError)?;
```

Comment:
Please double check my reasoning, but I think returning an `AggregateMetricsError` here causes the entire batch of buckets to fail: 
https://github.com/getsentry/relay/blob/043479fe570949003964c672ed3dda7253a98292/relay-metrics/src/aggregation.rs#L1134-L1143

Not sure what the best solution is, either swallow the error here or change `merge_all` to swallow the error there.

Reply:
ah right. fwiw that bogus code came from https://github.com/getsentry/relay/pull/1235/files#diff-9bce334fb404ce42d375eab0d36d6bb5d79f4d373c8c92c02f8a70dfea0cc1d8R1083

i'm gonna play around with this a bit and see